src/main/java/core/basesyntax/strategy/impl/OperationStrategyImpl.java

```
@@ -0,0 +1,29 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        if (operationHandlers == null) {
+            throw new RuntimeException("Operation handlers map cannot be null");
+        }
+        this.operationHandlers = operationHandlers;
```

Constructor only checks `operationHandlers != null` but does not verify the map contains handlers for each operation. The requirements state: "Strategy must validate that a handler exists for each operation and throw informative error if not." You should validate that for every `FruitTransaction.Operation` there is a non-null OperationHandler in the map. If any are missing, throw a RuntimeException naming the missing operations (include the enum names in the message) so the error is informative. Also consider performing this validation in the constructor (fail fast).